src/GafferSceneUI/LightPositionTool.cpp

```diff
@@ -0,0 +1,969 @@
+	// elaborate orientation calculation.
+
+	Context::Scope scopedContext( s.context() );
+
+	const M44f localTransform = s.scene()->transform( s.path() );
+
+	const M44f orientationMatrix = shadowSourceOrientation( s, shadowPivot, shadowTarget );
+
+	V3f originalRotation;
+	extractEulerXYZ( localTransform, originalRotation );
+	const M44f originalRotationMatrix = M44f().rotate( originalRotation );
+
+	const M44f relativeMatrix = originalRotationMatrix.inverse() * orientationMatrix;
+
+	V3f relativeRotation;
+	extractEulerXYZ( relativeMatrix, relativeRotation );
+
+	const V3f p = V3f( 0 ) * s.orientedTransform( Orientation::World );
+	const V3f offset = newP - p;
+
+	TranslationRotation( s ).apply( offset, relativeRotation );
+}
+
+bool LightPositionTool::affectsHandles( const Gaffer::Plug *input ) const
+{
+	if( TransformTool::affectsHandles( input ) )
+	{
+		return true;
+	}
+
+	return input == scenePlug()->transformPlug();
+}
+
+void LightPositionTool::updateHandles( float rasterScale )
+{
+	Selection s = selection().back();
+
+	handles()->setTransform( s.orientedTransform( Orientation::Local ) );
+
+	m_shadowHandle->setEnabled( selection().size() == 1 && TranslationRotation( s ).canApply() );
+
+	m_shadowHandle->setRasterScale( 0 );
+
+	auto shadowHandle = static_cast<ShadowHandle *>( m_shadowHandle.get() );
+
+	m_key = MurmurHash().append( s.path() );
+
+	std::optional<V3f> shadowPivot = getShadowPivot();
+	std::optional<V3f> shadowTarget = getShadowTarget();
+
+	shadowHandle->setShadowPivot( shadowPivot );
+	shadowHandle->setShadowTarget( shadowTarget );
+
+	if( !shadowPivot || !shadowTarget )
+	{
+		return;
+	}
+
+	// The user can control the distance along the line from shadow target
+	// to pivot, and the rotation around the Z-axis. Any variance from those
+	// contraints invalidates the stored shadow parameters.
+
+	Context::Scope scopedContext( s.context() );
+
+	const M44f worldTransform = s.scene()->fullTransform( s.path() );
+	const V3f p = worldTransform.translation();
+	const Line3f handleLine( shadowTarget.value(), shadowPivot.value() );
+
+	V3f direction;
+	worldTransform.multDirMatrix( V3f( 0, 0, -1.f ), direction );
+
+	const V3f handleDir = ( shadowTarget.value() - shadowPivot.value() ).normalized();
+
+	if(
+		!direction.equalWithAbsError( ( shadowTarget.value() - shadowPivot.value() ).normalized(), 1e-4 ) ||
+		handleLine.distanceTo( p ) > shadowHandle->getPivotDistance().value() * 1e-4
+	)
+	{
+		shadowHandle->setShadowPivot( std::nullopt );
+		shadowHandle->setShadowTarget( std::nullopt );
+		shadowHandle->setPivotDistance( std::nullopt );
+
+		m_shadowTargetMap.erase( m_key );
+		m_shadowPivotMap.erase( m_key );
+		m_shadowPivotDistanceMap.erase( m_key );
+	}
+
+	shadowHandle->setPivotDistance( ( p - shadowPivot.value() ).length() );
+}
+
+bool LightPositionTool::keyPress( const KeyEvent &event )
+{
+	// We track this regardless of whether we're active or not in case the tool
+	// is changed whilst the key is held down.
```

Comment:
I copied this from `RotateTool` without really thinking through the accuracy, but I believe in both cases the comment is incorrect and should be removed - tracking is in fact conditioned on active state. Or is my post-holiday brain not fully online and I'm mistaken now?

I don't think removing the condition is right since that would cause the cursor to change. Probably just need to remove the comment I think.

Reply:
Removing the comment looks good to me.

Reply:
Done in https://github.com/GafferHQ/gaffer/pull/5558/commits/d1d01f938346dd42dc12cbd3567ef70124a8784b